Docs to get started with Docker, Mobile and Web Test Environment setups

mkasireddy opened the pull request:
Documentation and guidelines to get started with Dockerized Mobile and Web Test Environment setups

Jira: https://invisionapp.atlassian.net/browse/SILVER-816

invisionjeremy commented:
@mkasireddy Can we add a section on how to add to the test suite?

Jorshasaur commented on getting_started_guide.md:
````diff
@@ -0,0 +1,51 @@
+-  To kick off Web tests from within a Docker container, on a local docker container:
+
+```shell
+    ./node_modules/.bin/wdio wdio.conf.local.js 
````
@mkasireddy is there a way we can run these commands from outside of the shell?  Maybe using npm?

mkasireddy replied:
@Jorshasaur Thats on my list of things to(simplify) look at

Jorshasaur commented on mobile_test_setup.md:
````diff
@@ -31,7 +29,7 @@ With `mocha`, you can use any options you feel like. But you will need to specif
 
 ```shell
-DEVICE=android:4.4 mocha -t 180000 -R spec test/functional/Mobile/ios/web-specs.js
+DEVICE=ios:9.3 mocha -t 260000 -R spec test/functional/Mobile/ios/web-specs.js
````
@mkasireddy is the `-t` command for the timeout?

mkasireddy replied:
@Jorshasaur yes. Im actually setting it as a global property. But when the set timeout isn't working Im overriding here

Jorshasaur commented on getting_started_guide.md:
```diff
@@ -0,0 +1,51 @@
+
+[To get setup with the Docker container Environment](https://github.com/InVisionApp/mobile-test-runner/blob/master/README.md)
+
+[To get started with Automated Mobile tests](https://github.com/InVisionApp/mobile-test-runner/blob/master/getting_started_guide.md)
```
@mkasireddy I think you've got the doc linking back to itself here, probably not what you intended?

mkasireddy replied:
ah good find @Jorshasaur thats not what I wanted to link.

datinc replied:
Seems like the link is 404

Jorshasaur commented:
@mkasireddy here are my suggestions:

1. I think we need a clearer step by step guide on how to write a test, especially with the web portion. 
2. A best practices section would also be useful, with your experience what rules should we follow to prevent us from writing brittle tests?
3. What would the process be for handling errors, especially if this runs remotely?  Would we review the video on sauce labs and go from there?  I think we'd need to figure out a process.

datinc commented:
Would be nice to have a readme or link to a webpage on how to create more tests. Did you use a framework to write the tests? If so could we have a link to have a look exposed API?

mkasireddy commented:
@Jorshasaur @juansalas @datinc @invisionjeremy 

 I've Modified the getting_started.md and added the Screencast that walks you thru the Test creation and Test Run process. Please take a look!

invisionjeremy commented on getting_started_guide.md:
```diff
@@ -1,24 +1,31 @@
--  Web automation is not within Mobile testing scope, but nevertheless we needed to Implement the Web Test framework utilising [Webdriver.io](http://webdriver.io/api.html) for a bunch of Web features.
+-  Appium API [here] (https://github.com/admc/wd/blob/master/doc/api.md)
+
+-  Get setup with Silver Team's Automated Mobile tests [Instructions are found here](https://github.com/InVisionApp/mobile-test-runner/mobile_test_setup.md)
```
I don't see a doc here @mkasireddy

Jorshasaur replied:
@invisionjeremy she hasn't merged it into master yet, but I see it at https://github.com/InVisionApp/mobile-test-runner/blob/SILVER-816/mobile_test_setup.md

invisionjeremy replied:
Ahh, I must have been looking at master. Thanks @Jorshasaur

invisionjeremy commented on getting_started_guide.md:
````diff
@@ -43,9 +50,16 @@
\ No newline at end of file
+```
+
+-  More detailed demo of creating an Automated Mobile-Web Test is explained in this [Screencast] (https://www.dropbox.com/s/502h2qm1pop4yk3/mobile_web_auto_demo.mov?dl=0)
````
@mkasireddy The screencast looks good. I cleaned up the audio and minimized the file here: Mobile Test Screencast-2.mp4

invisionjeremy commented:
Looks good Manju 👍

mkasireddy commented:
Thanks @invisionjeremy ! Merging away...
